src/preferences.c

```diff
@@ -108,22 +98,30 @@ void preferences_apply(GtkWidget *widget, gpointer data)
    }
 }
 
-
-void preferences_ok(GtkWidget *widget, gpointer data) {
-   GtkWidget **buttons = data;
-   preferences_apply(widget, buttons);
-   gtk_widget_destroy(buttons[PR_DIALOG]);
+static void prefs_dialog_response (GtkDialog * dlg, int response, gpointer user_data)
+{
+   switch (response)
+   {
+      case GTK_RESPONSE_APPLY:
+         preferences_apply (user_data);
+         g_signal_stop_emission_by_name (dlg, "response");
+         return;
+         break;
```

Comment:
Is `break` here necessary? My C knowledge is very limited but it's a no-op here and as far as I remember C doesn't require every switch hand ending with `break` statement.